Wrappers/Python/cil/processors/Masker.py

```diff
@@ -153,11 +174,10 @@ def process(self, out=None):
         except:
             mask_arr = self.mask
 
-        try:
-            mask_invert = ~mask_arr
-        except TypeError:
-            raise TypeError("Mask expected to be a boolean array got {}".format(mask_arr.dtype))
-
+        mask_arr = numpy.array(mask_arr, dtype=bool)
+        
+        mask_invert = ~mask_arr
```

Comment:
Potentially this will change the values on the object provided by the user: you are changing its type.

```suggestion
        mask_invert  = numpy.array(mask_arr, dtype=bool)
        numpy.invert(mask_invert, out=mask_invert)
```

Could you add a unit test to see whether the mask remains untouched?

Reply:
Why would it change the values?

Reply:
https://numpy.org/doc/1.20/reference/generated/numpy.array.html